src/rebar_prv_plugins.erl

```diff
@@ -41,22 +42,33 @@ do(State) ->
 
     RebarOpts = rebar_state:opts(State),
     SrcDirs = rebar_dir:src_dirs(RebarOpts, ["src"]),
-    Plugins = rebar_state:get(State, plugins, []),
-    ProjectPlugins = rebar_state:get(State, project_plugins, []),
+    {LocalPluginsDefs, _} = list_local_plugins(State),
     PluginsDirs = filelib:wildcard(filename:join(rebar_dir:plugins_dir(State), "*")),
 
     %% use `checkouts_dir' and not `checkouts_out_dir'. Since we use `all' in `find_apps'
     %% so it doesn't need to be built and the apps in `checkouts_dir' could be old
     %% because the user removing from `_checkouts/' doesn't cause removal of the output
     CheckoutsDirs = filelib:wildcard(filename:join(rebar_dir:checkouts_dir(State), "*")),
     Apps = rebar_app_discover:find_apps(CheckoutsDirs++PluginsDirs, SrcDirs, all, State),
-    display_plugins("Local plugins", Apps, Plugins ++ ProjectPlugins),
+    display_plugins("Local plugins", Apps, LocalPluginsDefs),
     {ok, State}.
 
 -spec format_error(any()) -> iolist().
 format_error(Reason) ->
     io_lib:format("~p", [Reason]).
 
+list_local_plugins(State) ->
+    LocalPluginsDefs = rebar_state:get(State, plugins, [])
+                       ++ rebar_state:get(State, project_plugins, []),
+    LocalPluginsNames = lists:map(
+                            fun (LocalPluginDef) ->
+                                if is_atom(LocalPluginDef) -> LocalPluginDef;
+                                   is_tuple(LocalPluginDef) -> element(1, LocalPluginDef)
```

Comment:
is the type of `element(1, LocalPluginDef)` going to make sense? in `display_plugins`, the first element is turned to a `binary` in all cases, whereas we get `atom_to_binary` as a forcing one there. I assume it should be okay (no reason to get lists in plugin names), and Dialyzer would otherwise catch it.

Reply:
Hm...

> is the type of element(1, LocalPluginDef) going to make sense?

I assumed `atom()` would be appropriate, here, but if the input is not always one, it would break, sure. Do you prefer I create+use a `rebar_utils:to_atom(MaybeNonAtom)` fun.?

Reply:
Yeah, I think it would be safer. I don't really expect a non-atom either, but there are weird things sometimes happening so the broader the safer. Could be a list in there at some point :/

Reply:
OK. Only lists and atoms, then?

Reply:
the utils to_atom essentially does `binary | list | atom -> atom`, which is just a catch all "whatever the hell we passed, turn it to an atom". It's generally a sort of anti-pattern, but if we relied on it already, I'm hesitant to clamp down on the types without risking rogue errors.

Reply:
10404ab